llvm/lib/CodeGen/LiveDebugValues/VarLocBasedImpl.cpp

```diff
@@ -1811,6 +1818,11 @@ void VarLocBasedLDV::recordEntryValue(const MachineInstr &MI,
   VarLoc EntryValLocAsBackup = VarLoc::CreateEntryBackupLoc(MI, LS, NewExpr);
   LocIndex EntryValLocID = VarLocIDs.insert(EntryValLocAsBackup);
   OpenRanges.insert(EntryValLocID, EntryValLocAsBackup);
+
+  // In Swift async functions entry values are preferred, since they
+  // can be evaluated in both live frames and virtual backtraces.
+  if (isSwiftAsyncContext(MI))
+    MI.getOperand(3).setMetadata(NewExpr);
```

Comment:
I don't think we really want to create `EntryValLocAsBackup` if we're changing the MI's expression (as the backup would be identical to the original VarLoc).

Wdyt of making this change before the following sequence in `ExtendRanges`:

```
  // Only in the case of entry MBB collect DBG_VALUEs representing
  // function parameters in order to generate debug entry values for them.
  MachineBasicBlock &First_MBB = *(MF.begin());
  for (auto &MI : First_MBB) {
    collectRegDefs(MI, DefinedRegs, TRI);
      if (MI.isDebugValue())
        recordEntryValue(MI, DefinedRegs, OpenRanges, VarLocIDs);
  }
```

I.e., rewrite the expressions for all the swift async entry value MI's as the first step.

Then you can skip `recordEntryValue` for swift async MI's entirely.